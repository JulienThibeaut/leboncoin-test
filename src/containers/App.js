import React, { Component } from 'react';
import data from './../data/data'
import MessagesList from './../components/MessagesList'
import AddMessageForm from './../components/AddMessageForm'

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      messages: data.messages,
      text: '',
      visibility: null
    }
    this.addMessage = this.addMessage.bind(this)
  }

  addMessage = e => {
    const messages = data.messages
    messages.push({
      index: messages.length+1,
      text: this.state.text,
      visibility: this.state.visibility
    })
    this.setState({messages})
    e.preventDefault()
  }

  render() {
    const { text, visibility, messages } = this.state
    return (
      <div className="app">
        <h1>Leboncoin test</h1>
        <MessagesList messages={messages} />
        <AddMessageForm
          onSubmit={this.addMessage}
          handleText={(e) => this.setState({ text: e.target.value })}
          handleVisibility={(e) => this.setState({visibility: !this.state.visibility})}
          text={text}
          visibility={visibility} />
      </div>
    );
  }
}

export default App;
