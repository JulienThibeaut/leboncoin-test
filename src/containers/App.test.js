import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

describe('Component: App', () => {
  const messages = ['leboncoin', 'test', 'test technique'];

  it('should match its empty snapshot', () => {
    const tree = renderer.create(
      <App />
     ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should add an messages based on the value in the state', () => {
      const component = shallow(<App />);
      const preventDefault = jest.fn();
      component.setState({
        messages
      });
      component.find('form').simulate('submit', { preventDefault });
      expect(toJson(component)).toMatchSnapshot();
      expect(preventDefault).toBeCalled();
    });

});
