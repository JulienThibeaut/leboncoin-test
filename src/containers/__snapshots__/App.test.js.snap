// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: App should match its empty snapshot 1`] = `
<div
  className="app"
>
  <h1>
    Leboncoin test
  </h1>
  <ul
    className="message-list"
  >
    <li
      className="message"
    >
      <p>
        Sed fruatur sane hoc solacio atque hanc insignem ignominiam, quoniam uni praeter se inusta sit, putet esse leviorem, dum modo, cuius exemplo se consolatur, eius exitum expectet, praesertim cum in Albucio nec Pisonis libidines nec audacia Gabini fuerit ac tamen hac una plaga conciderit, ignominia senatus.
      </p>
      <p
        className="visibility"
      >
        private
      </p>
    </li>
    <li
      className="message"
    >
      <p>
        Post quorum necem nihilo lenius ferociens Gallus ut leo cadaveribus pastus multa huius modi scrutabatur. quae singula narrare non refert, me professione modum, quod evitandum est, excedamus.
      </p>
      <p
        className="visibility"
      >
        private
      </p>
    </li>
    <li
      className="message"
    >
      <p>
        Nunc vero inanes flatus quorundam vile esse quicquid extra urbis pomerium nascitur aestimant praeter orbos et caelibes, nec credi potest qua obsequiorum diversitate coluntur homines sine liberis Romae.
      </p>
      <p
        className="visibility"
      >
        public
      </p>
    </li>
  </ul>
  <form
    className="add-message"
    onSubmit={[Function]}
  >
    <label>
      <textarea
        id="text"
        onChange={[Function]}
        placeholder="Message"
        value=""
      />
    </label>
    <div
      className="end-form"
    >
      <label>
        Visibility:
        <a
          onClick={[Function]}
        >
           
          public
        </a>
      </label>
      <input
        className="btn"
        type="submit"
        value="Add message"
      />
    </div>
  </form>
</div>
`;
